lib/series_arithmetic.rb

```diff
@@ -119,7 +121,9 @@ def percentage_change
     new_series_data = {}
     last = nil
     data.sort.each do |date, value|
-      new_series_data[date] = (value-last)/last*100 unless last.nil?
+      if last && (last != 0 || value == 0)
+        new_series_data[date] = last == 0 ? 0 : (value - last) / last * 100
```

Comment:
There is a whole lot of nested logic going on here. Here's one alternative that is way longer but may be quicker to reason about:
```
def percent_change(value, last)
  if last.nil?
    return nil
  end
  if last == 0 && value !== 0
    return nil
  end
  if last == 0 && value == 0
    return 0
  end
  return (value - last) / last * 100
end
```

Reply:
Point taken.